pkgs/development/libraries/imlib2/default.nix

```diff
@@ -1,14 +1,16 @@
 { lib, stdenv, fetchurl
 # Image file formats
-, libjpeg, libtiff, giflib, libpng, libwebp
+, libjpeg, libtiff, giflib, libpng, libwebp, libjxl
+, libspectre
 # imlib2 can load images from ID3 tags.
 , libid3tag, librsvg, libheif
 , freetype , bzip2, pkg-config
 , x11Support ? true, xlibsWrapper ? null
-# Compilation error on Darwin with librsvg. For more information see:
-# https://github.com/NixOS/nixpkgs/pull/166452#issuecomment-1090725613
-, svgSupport ? !stdenv.isDarwin
-, heifSupport ? !stdenv.isDarwin
+, svgSupport ? false
+, heifSupport ? false
+, webpSupport ? false
```

Comment:
Set of default flags should not be minimal but rather what is useful to majority of reverse dependencies. E.g. if a music player uses imlib2 to render album art obtained from web, I would expect it to require imlib2 compiled with webp support. Or application launcher will need SVG support to render app icons.

Reply:
Sure, I was just basing it on the fact that support for each of these formats was only added in the last few months (also by @nagy) so clearly everyone coped well enough without these for years before now :shrug: 

Do say if you specifically think one or more of these should be enabled by default.

Also, there's nothing stopping an **application** choosing to use `imlib2Full` if it has a specific need for one of these formats.

Reply:
To be honest, I would rather just make the full variant the default. As mentioned above, all those libraries will already be part of the system closure of any desktop system. And at that point the minimal variant does not make much sense.

It is hard to tell whether the apps were subtly broken – e.g. if some album art in a music player does not show, the user might just assume it is not available, rather than that it is in webp format and there is a missing dependency.

The point of an image library should be that it can open images and as such, I would say that mere 200 MB increase (most likely amortized) is worth it.

But I am not familiar with imlib2 specifically – maybe each app needs to explicitly declare which formats it wants to use, in which case adding more dependencies would not really do anything.

Reply:
That might be fine for the desktop use case, but it means people building docker images for server deployment, where they possibly only ever need to work with one or two common formats end up with _huge_ containers. While  desktop applications are free to use `imlib2Full`.

Reply:
We could switch the GUI desktop applications over to the full variant with this PR.